tests/describe_task_queue_test.go

```diff
@@ -0,0 +1,230 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tests
+
+import (
+	"context"
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/testing/historyrequire"
+	"go.temporal.io/server/common/testing/protorequire"
+	"go.temporal.io/server/common/testing/updateutils"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+// making a new test suite here
+
+type (
+	DescribeTaskQueueSuite struct {
+		*require.Assertions
+		protorequire.ProtoAssertions
+		historyrequire.HistoryRequire
+		updateutils.UpdateUtils
+		FunctionalTestBase
+	}
+)
+
+func (s *DescribeTaskQueueSuite) SetupSuite() {
+	s.dynamicConfigOverrides = map[dynamicconfig.Key]any{
+		dynamicconfig.FrontendEnableWorkerVersioningDataAPIs:     true,
+		dynamicconfig.FrontendEnableWorkerVersioningWorkflowAPIs: true,
+		dynamicconfig.FrontendEnableWorkerVersioningRuleAPIs:     true,
+		dynamicconfig.MatchingForwarderMaxChildrenPerNode:        partitionTreeDegree,
+		dynamicconfig.TaskQueuesPerBuildIdLimit:                  3,
+
+		dynamicconfig.AssignmentRuleLimitPerQueue:              10,
+		dynamicconfig.RedirectRuleLimitPerQueue:                10,
+		dynamicconfig.RedirectRuleChainLimitPerQueue:           10,
+		dynamicconfig.MatchingDeletedRuleRetentionTime:         24 * time.Hour,
+		dynamicconfig.ReachabilityBuildIdVisibilityGracePeriod: 3 * time.Minute,
+		dynamicconfig.ReachabilityQueryBuildIdLimit:            4,
+
+		// Make sure we don't hit the rate limiter in tests
+		dynamicconfig.FrontendGlobalNamespaceNamespaceReplicationInducingAPIsRPS:                1000,
+		dynamicconfig.FrontendMaxNamespaceNamespaceReplicationInducingAPIsBurstRatioPerInstance: 1,
+		dynamicconfig.FrontendNamespaceReplicationInducingAPIsRPS:                               1000,
+
+		// The dispatch tests below rely on being able to see the effects of changing
+		// versioning data relatively quickly. In general, we only promise to act on new
+		// versioning data "soon", i.e. after a long poll interval. We can reduce the long poll
+		// interval so that we don't have to wait so long.
+		// TODO: update this comment. it may be outdated and/or misleading.
+		dynamicconfig.MatchingLongPollExpirationInterval: longPollTime,
+
+		// this is overridden for tests using testWithMatchingBehavior
```

Comment:
I think you should be able to get rid of all these lines. (We can add more config overwrites when some tests in this file need them)